### PR TITLE
Add soxr recipe

### PR DESCRIPTION
**Pull request by HosameldinMohamed**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by HosameldinMohamed** [comment]
Enabled OpenMP on `linux` and `macOS` by adding the `libgomp` and `llvm-openmp` packages to the build section of the `meta.yaml` file

https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp

**Comment by HosameldinMohamed** [comment]
It fails because it doesn't find the `LICENSE` file.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=479682&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1537

In `meta.yaml` I put 

```
  license_file: ${PREFIX}/share/doc/libsoxr/LICENCE
```

Because I saw the `LICENSE` file is installed in `<CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/LICENCE`

**Comment by HosameldinMohamed** [comment]
Same problem is related to the test files, here are the files I need to test: (for linux)

```yaml
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/README  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/LICENCE  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/NEWS  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/lib/libsoxr.so.0.1.2  # [linux]
    - test -f <CMAKE_INSTALL_PREFIX>/lib/libsoxr.so.0  # [linux]
    - test -f <CMAKE_INSTALL_PREFIX>/lib/libsoxr.so  # [linux]
    - test -f <CMAKE_INSTALL_PREFIX>/include/soxr.h  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/lib/libsoxr-lsr.so.0.1.9  # [linux]
    - test -f <CMAKE_INSTALL_PREFIX>/lib/libsoxr-lsr.so.0  # [linux]
    - test -f <CMAKE_INSTALL_PREFIX>/lib/libsoxr-lsr.so  # [linux]
    - test -f <CMAKE_INSTALL_PREFIX>/include/soxr-lsr.h  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/examples/1-single-block.c  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/examples/1a-lsr.c  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/examples/2-stream.C  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/examples/3-options-input-fn.c  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/examples/4-split-channels.c  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/examples/5-variable-rate.c  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/examples/examples-common.h  # [not win]
    - test -f <CMAKE_INSTALL_PREFIX>/share/doc/libsoxr/examples/README  # [not win]
   ```

**Comment by traversaro** [comment]
@HosameldinMohamed I think it is sufficient to just test one header file and one library file (two on Windows if necessary).

**Comment by HosameldinMohamed** [comment]
Error in building Windows

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=479699&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=06d62fd2-ab49-5add-9ee5-83b6882d7d2b&l=1563

```
See also "C:/bld/soxr_1648042412722/work/build/CMakeFiles/CMakeOutput.log".
    NMake Makefiles

  does not support platform specification, but platform

    x64

  was specified.
```

**Comment by traversaro** [comment]
Just FYI you can also use Ninja on Windows for uniformity.

**Comment by HosameldinMohamed** [comment]
I added the tests but the checks are not running with the new commits anymore.

**Comment by HosameldinMohamed** [comment]
> Just FYI you can also use Ninja on Windows for uniformity.

At the end I compiled Windows and non-windows with `make`.

**Comment by HosameldinMohamed** [comment]
@conda-forge/staged-recipes, ready for review, thanks.

**Comment by traversaro** [comment]
> @conda-forge/staged-recipes, ready for review, thanks.

You are a new user so you can't mention, I will give a first round of review and then mention it for you.

File: recipes/soxr/bld.bat [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+:: Build.
+cmake --build . --config Release
+if errorlevel 1 exit 1
+
+:: Install.
+cmake --build . --config Release --target install
+if errorlevel 1 exit 1
+
+:: Test.
+ctest --output-on-failure -C Release -E "misc::check_license"
```
Comment (traversaro) [inline recipes/soxr/bld.bat:21]: Unless there is a `misc::check_license` test, it does not make sense to try to skip a test name like that (the `-E` option of ctest just excluds the test that contain `misc::check_license` in their name).

File: recipes/soxr/build.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+mkdir build
+cd build
+
+cmake ${CMAKE_ARGS}  .. \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DBUILD_TESTS=ON
+
+cmake --build . --config Release
+cmake --build . --config Release --target install
+ctest --output-on-failure -C Release -E "misc::check_license"
```
Comment (traversaro) [inline recipes/soxr/build.sh:12]: See other comment.

File: recipes/soxr/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  summary: 'The SoX Resampler library'
+  description: |
+    High quality, one-dimensional sample-rate conversion library
+
+  doc_url: https://sourceforge.net/p/soxr/wiki/Home/
+  dev_url: https://sourceforge.net/projects/soxr/
+
+extra:
+  recipe-maintainers:
+    - HosameldinMohamed
```
Comment (traversaro) [inline recipes/soxr/meta.yaml:48]: I am happy to be a mantainer as well if it is ok for you!

File: recipes/soxr/bld.bat (hunk 1 above)
Comment (HosameldinMohamed) [reply recipes/soxr/bld.bat:21]: sorry I copied the file from another receipt..

There's no test with this name, I'll remove it!

File: recipes/soxr/meta.yaml (hunk 3 above)
Comment (HosameldinMohamed) [reply recipes/soxr/meta.yaml:48]: That would be great!
I'll add you!

**Comment by HosameldinMohamed** [comment]
@traversaro I addressed your comments!

**Comment by traversaro** [comment]
Thanks! 

https://github.com/orgs/conda-forge/teams/staged-recipes @conda-forge/help-c-cpp The recipe is ready for review.

File: recipes/soxr/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    - {{ compiler('c') }}
+    - make
+    - llvm-openmp  # [osx]
+    - libgomp  # [linux]
+  host:
+    - tinyxml
+
+test:
+  commands:
+    - test -f ${PREFIX}/include/soxr.h  # [not win]
```
Comment (carterbox) [inline recipes/soxr/meta.yaml:28]:
```suggestion
    - test -f ${PREFIX}/include/soxr.h  # [unix]
```

**Review by carterbox — CHANGES_REQUESTED** [review]
[Conda thinks that you aren't dynamically linking against tinyxml or libstdcxx-ng/libcxx/vsruntime for each of the three builds.](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=480287&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1534) Is that the case? It doesn't seem like tinyxml is a header-only library, so conda should be able to detect the links, and you shouldn't be statically linking to tinyxml. The runtime libraries you can add to the ignore_run_exports section because you may not be using

**Comment by HosameldinMohamed** [comment]
> [Conda thinks that you aren't dynamically linking against tinyxml or libstdcxx-ng/libcxx/vsruntime for each of the three builds.](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=480287&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1534) Is that the case? It doesn't seem like tinyxml is a header-only library, so conda should be able to detect the links, and you shouldn't be statically linking to tinyxml. The runtime libraries you can add to the ignore_run_exports section because you may not be using the standard libraries.

Thanks @carterbox for your checks.

I added

```yaml
  ignore_run_exports:
    - libstdc++
```

[But the warning is still persisting](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=488061&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1484), I'll try to ignore the correct package.

I also realized that `tinyxml` is not required at all! My bad!

@traversaro

**Comment by carterbox** [comment]
```yaml
  ignore_run_exports_from:
    - {{ compiler('cxx') }}
```
I recently learned about `ignore_run_exports_from`. You could try that. It may be more robust.

**Comment by isuruf** [comment]
Why not remove the C++ compiler from build instead?

File: recipes/soxr/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+build:
+  number: 0
+  ignore_run_exports:
+    - libstdcxx-ng
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('cxx') }}
```
Comment (traversaro) [inline recipes/soxr/meta.yaml:20]:
```suggestion
```

As suggested by @isuruf

**Comment by traversaro** [comment]
> @traversaro

See https://github.com/conda-forge/staged-recipes/pull/17935/files#r845431275 .

File: recipes/soxr/meta.yaml (hunk 5 above)
Comment (HosameldinMohamed) [reply recipes/soxr/meta.yaml]: done in https://github.com/conda-forge/staged-recipes/pull/17935/commits/e315dc9fce53e45c4c4113d4c9baea7f94cb6fb7

**Comment by HosameldinMohamed** [comment]
Thanks!

**Pull request merged**